### ISSUE TITLE
SEO for Astropy

### ISSUE THREAD

**Issue by embray** [astropy/astropy#3830]
Two current methods for searching for Python packages don't place Astropy very highly.  The first is:

```
$ pip search astronomy
```

which for me currently results in:

```
astropyp                  - Astronomy Pypeline Framework and FITS Viewer
image_registration        - Image Registration Tools for extended images in astronomy.
novas                     - The United States Naval Observatory NOVAS astronomy library
skyfield                  - Elegant astronomy for Python
gammapy                   - A Python package for gamma-ray astronomy
astroobs                  - Provides astronomy ephemeris to plan telescope observations
PyAstronomy               - A collection of astronomy related tools for Python.
astroML                   - tools for machine learning and data mining in Astronomy
Barak                     - A set of astronomy-related routines for generating Voigt profiles from atomic data,
                            reading and writing data, working with SEDs, passbands and dust extinction laws.
astropy                   - Community-developed python astronomy tools
gary                      - Galactic astronomy and gravitational dynamics.
palpy                     - PAL -- A Positional Astronomy Library
python-casacore           - A wrapper around CASACORE, the radio astronomy library
```

(as an aside, are we sure we don't want to name the Astropy distribution "Astropy", i.e. capitalized instead of lowercase?  I think it would stand out more.)

```
$ pip install astrophysics
```

fares even worse:

```
Astropysics               - Astrophysics libraries for Python
BOTEC                     - A simple astrophysics and orbital mechanics simulator
gwpy                      - A package to enable gravitational-wave astrophysics in Python.
```

(although at least we get Astropysics, which points to Astropy in its docs, but Astropy should be searchable under "astrophysics" too IMO).

I'm not sure how pip chooses how to order search results, but I'll look into it and see if we can "game" the system :)

The search bar on PyPI is no better:
![astropy-pypi](https://cloud.githubusercontent.com/assets/676149/8065629/a0f3779e-0eb0-11e5-8997-69d488f1afd2.png)
This at least displays all current versions of Astropy, so it stands out more.  But it's still a bit silly.  It also doesn't rank the versions in any useful way, and in call cases they are given a "Weight" of 3, where according to the site "Weight" is defined by "Occurrence of search term weighted by field (name, summary, keywords, description, author, maintainer)".  PyAstronomy wins out in this case since "astronomy" is right in the name.

Astropy _does_ show up on PyPI with a search for "astrophysics", but with a Weight of 1, below a number of other packages (the term "astrophysics" does appear once in the package description).

**Comment by embray** [comment]
By the way, I bring this up because as of https://bugs.python.org/issue23891, the Python official documentation finally features some documentation for current best practices, including use of virtualenv and pip.  There is an [example](https://hg.python.org/cpython/rev/15ee0e7078e3#l2.102) for pip demonstrating the result of "`pip search astronomy`".  But the results are cut off early, before astropy.  So we lost out on an unexpected opportunity to be featured in the official Python docs :(

**Comment by astrofrog** [comment]
See information here about how the weight is calculated: http://stackoverflow.com/questions/28685680/what-does-weight-on-search-results-in-pypi-help-in-choosing-a-package

**Comment by bsipocz** [comment]
Assigning myself and will try to look into how we can improve this. Even my dummy gastropy spaceholder comes up sooner than astropy 😞

**Comment by pllim** [comment]
This is what we are getting in 2019:
```
astronomy (0.0.1)           - Astronomy!
gastropy (0.0dev)           - (g)astronomy
n2-tools (0.0.24)           - tools for astronomy data
skyfield (1.10)             - Elegant astronomy for Python
celespy (0.0.4)             - Generic astronomy functions
intake-astro (0.1.1)        - Astronomy plugins for Intake
astro-scripts (0.4.0)       - Small scripts for astronomy
palpy (1.8.1)               - PAL -- A Positional Astronomy Library
numina (0.20.1)             - Astronomy data reduction library
codex-africanus (0.1.4)     - Radio Astronomy Building Blocks
acalib (0.1.3)              - Advanced Computing for Astronomy Library
sinistra (0.3.3)            - A collection of astronomy related tools.
gary (0.1a)                 - Galactic astronomy and gravitational dynamics.
hips (0.2)                  - Python astronomy package for HiPS
astropy (3.1.2)             - Community-developed python astronomy tools
  INSTALLED: 3.2.dev23959
  LATEST:    3.1.2
astrotoyz (0.1.4)           - Astronomy tools built on the Toyz framework
starry (0.3.0)              - Analytic occultation light curves for astronomy.
experimentsLogReader (1.2)  - Log file reader for Radio Astronomy
mclearn (0.1.6)             - Active learning algorithms with application in astronomy.
gammapy (0.10)              - A Python package for gamma-ray astronomy
pyastroschema (0.5.4)       - Standardized specifications for astronomy and astrophysics data.
fidia (0.4rc1)              - Format Independent Data Interface for Astronomy
ragavi (0.0.4)              - Radio Astronomy Gain and Visibility Inspector
PyAstronomy (0.13.0)        - A collection of astronomy related tools for Python.
astroobs (1.4.5)            - Provides astronomy ephemeris to plan telescope observations
DDFacet (0.3.4.1)           - Facet-based radio astronomy continuum imager
astroML (0.4.post1)         - tools for machine learning and data mining in Astronomy
astrodbkit (0.6.6)          - Astronomy database management using SQL and Python
vaex-astro (0.3.0)          - Astronomy related transformations and FITS file support
pyzpace (0.0)               - Zach Pace's astronomy-related python tools
python-casacore (3.0.0)     - A wrapper around CASACORE, the radio astronomy library
cadctap (0.8)               - Client for accessing data at the Canadian Astronomy Data Centre
astrocats (0.3.37)          - Package for downloading, analyzing, and constructing open astronomy
                              catalogs.
novas (3.1.1.4)             - The United States Naval Observatory NOVAS astronomy library
simspeclib (2.3.0.3)        - SimSpec library. SIMplified SPECtrum reduction package for
                              astronomy.
cadcetrans (0.9.2)          - Client for transferring data Canadian Astronomy Data Centre
cadcdata (1.3.1)            - Client for accessing data at the Canadian Astronomy Data Centre
simspec-lib (2.4.0.1)       - SimSpec library. SIMplified SPECtrum reduction package for
                              astronomy.
f311 (18.3.1.0)             - Astronomy-related API, command-line tools, and windowed applications
astrobase (0.4.0)           - Python modules and scripts useful for variable star work in
                              astronomy.
astromodels (0.5.1)         - Astromodels contains models to be used in likelihood or Bayesian
                              analysis in astronomy
simspec (1.8.5)             - User interface for SimSpec. SIMplified SPECtrum reduction package
                              for astronomy.
cygrid (1.0.1)              - Cygrid is a cython-powered convolution-based gridding module for
                              astronomy
SWHT (0.1.2)                - Spherical Wave Harmonic Transform for radio astronomy visibility
                              data
monashspa (1.3.0)           - Library of useful data analysis tools for Monash University Physics
                              & Astronomy students
nmmn (0.8.4)                - Miscellaneous methods for astronomy, dealing with arrays,
                              statistical distributions and computing goodness-of-fit
thefriendlystars (0.0.2)    - Python toolkit for managing catalogs of stars including tools for
                              querying popular astronomy archives.
astropy-stark (1.0.0)       - random-custom astronomy functions including making fake accretion
                              disk light curves
zogyp (1.4.5)               - A parallel version of ZOGY image subtraction for astronomy, with
                              Proper Co-Addition and Image alignment
Barak (0.3.2)               - A set of astronomy-related routines for generating Voigt profiles
                              from atomic data, reading and writing data, working with SEDs,
                              passbands and dust extinction laws.
```

**Comment by zdroid** [comment]
@pllim I got even worse results trying this in 2020. This looks like a design flaw in pip / PyPI and not a mistake on this project's side. Compared to npmjs.org search for example, PyPI search is quite terrible.

**Comment by taldcroft** [comment]
Unrelated to pypi, but if I go to duckduckgo or bing (less biased than google for me), searching on "python astronomy" goes get astropy.org in the first few, but #1 is http://www.astropython.org/.  This site was a nice idea at the time but it's basically inactive.  I've been thinking about shutting it down, but maybe we can repurpose the URL for something more useful?

**Comment by zdroid** [comment]
The purpose of the website is quite great, to serve as a knowledge base for the astronomy software. Perhaps if enough people were interested in continuing the project, it could be re-started. (and I wouldn't limit it to Python only, as there's software like Stellarium and Celestia written in C++)

**Comment by pllim** [comment]
Re: astropython.org -- If you include C++, then the URL would be misleading, no? 😬

**Comment by taldcroft** [comment]
What the astropython.org site really needs is contributors and maintainers.  Sadly both are lacking.

**Comment by zdroid** [comment]
Perhaps pushing https://github.com/astropy/astropython.org into prominence could move the project out of the obscurity. It's pretty much invisible compared to all other repositories under astropy. astropy/astropy has quite a few contributors, so astropython.org could have them, too.

**Comment by bsipocz** [comment]
@zdroid - as @taldcroft said, astropython.org was a proof of concept that didn't really take off as people are already spread very thin.

On the other hand, if you have ideas how to improve the astropy's SEO, we would be more than happy to make adjustments.
E.g. I think we should change the `Community-developed python astronomy tools` blurb for astropy on pypi to be shorter and more focused. e.g. `Community core library for astronomy` or similar should get higher score. After all our one word described joke packages are at the top of that list....

**Comment by zdroid** [comment]
I could find only this code from [pypa/warehouse](https://github.com/pypa/warehouse/blob/dc120b9c0f8fba9dc98809e68e1b56a1d1bda5e6/warehouse/search/queries.py) that reliably sets the order of packages:

```python
SEARCH_BOOSTS = {
    "name": 10,
    "normalized_name": 10,
    "description": 5,
    "keywords": 5,
    "summary": 5,
}
```

It's understandable that `astronomy` and `gastronomy` will be on the top. The reasoning behind other packages that come up after is very unclear. Maybe putting `astronomy` at the beginning or at the end gives a higher score than putting it somewhere in the middle. We could try that anyway, for example in the way you suggested (`Community core library for astronomy`).

**Comment by bsipocz** [comment]
Would you be interested contributing the change in a PR?

**Comment by zdroid** [comment]
@bsipocz I opened a PR for that, so there we can continue the discussion about which description is the most suited.

**Comment by zdroid** [comment]
SEO isn't bad only on PyPI, it's bad on GitHub too. Searching for astronomy and astrophysics doesn't put Astropy highly. I suggest this repository's description being changed to "Astronomy and astrophysics core library" too, and `astrophysics` being added to repository's tags.

**Comment by pllim** [comment]
Re: Changing repo description -- I think only @astropy/coordinators can do that.

**Comment by hamogu** [comment]
I've adjusted the description and added more tags on github. Search on bing/duckduckgo seems reasonable for me (at least today). Pypi is still terrible, but I don't think there is much more we can do about it. Also, fortunately, astropy is pretty well known in the community and comes up on top for all of the more specific searches I tried (e.g. "read fits file python" or "python WCS". 

Since it seems that we've done what we can reasonably do (while the pypi search results are still terrible), should we close the issue?
